src/main/java/teammates/ui/webapi/DeleteAccountAction.java

```diff
@@ -1,19 +1,39 @@
 package teammates.ui.webapi;
 
+import com.google.firebase.auth.AuthErrorCode;
+
+import teammates.common.exception.AuthException;
 import teammates.common.util.Const;
+import teammates.common.util.Logger;
 
 /**
  * Action: deletes an existing account (either student or instructor).
+ * <p>The associated Firebase user is also deleted.</p>
  */
 class DeleteAccountAction extends AdminOnlyAction {
 
+    private static final Logger log = Logger.getLogger();
+
     @Override
-    public JsonResult execute() {
+    public JsonResult execute() throws InvalidOperationException {
         String googleId = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
         if (fileStorage.doesFileExist(googleId)) {
             fileStorage.delete(googleId);
         }
         logic.deleteAccountCascade(googleId);
+
+        try {
+            authProxy.deleteUser(googleId);
+        } catch (AuthException e) {
+            if (AuthErrorCode.USER_NOT_FOUND.toString().equals(e.getErrorCode())) {
+                // Deleting Firebase user error of type user not found logged as warning and not thrown as exception
+                // to reduce unnecessary attention as old TEAMMATES users are not Firebase users
+                log.warning("Firebase user not found: " + e.getMessage());
+            } else {
+                throw new InvalidOperationException("Cannot delete Firebase user", e);
```

Comment:
`InvalidOperationException` translates to 409. This may be a 5XX error.

Reply:
Is changing it to a `RuntimeException` appropriate?

Reply:
I have shifted the logic for checking if exceptions are thrown due to non-existence to the `deleteUser()` method in `FirebaseAuthService.java` itself. Hence, any other exceptions thrown will result in an `AuthException` and eventually a `500 Internal Server Error` in `WebApiServlet.java`.